src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -574,7 +574,9 @@ private ProtocolToken ProcessBlob(int frameSize)
 
                     frameSize = nextHeader.Length + TlsFrameHelper.HeaderSize;
                     // Can process more handshake frames in single step, but we should avoid processing too much so as to preserve API boundary between handshake and I/O.
-                    if ((nextHeader.Type != TlsContentType.Handshake && nextHeader.Type != TlsContentType.ChangeCipherSpec) || frameSize > _handshakeBuffer.ActiveLength)
+                    // TODO: Fix this check: Accept AppData type with Handshake content type
```

Comment:
This would be encrypted, right? This is purely per optimization but I would expect it does not impact any functionality, right?

Reply:
Yes, we can't check the inner content type, I wrote the comment before I thought this completely through.

I did not observe any test failures regarding the commented condition. So it should be fine, but we can probably still keep the optimization by checking if we are currently attempting "renegotiation".

Reply:
added check of `_renego` flag, seems to work fine